src/utils/work/seekWorkResponse.js

```diff
@@ -5,34 +5,43 @@ import fetchAPI from 'utils/fetchAPI';
 import unpackResult from 'utils/work/unpackResult';
 import WorkResponseError from 'utils/WorkResponseError';
 
+const throwResponseError = (response) => {
+  throw new Error(`Error ${response.status}: ${response.text}`, { cause: response });
+};
+
 const seekFromS3 = async (ETag, experimentId) => {
   const response = await fetchAPI(`/v1/workResults/${experimentId}/${ETag}`);
-  const { signedUrl } = await response.json();
 
-  if (!signedUrl) return null;
+  if (!response.ok) {
+    if (response.status === 404) return null;
+
+    throwResponseError(response);
+  }
+
+  const { signedUrl } = await response.json();
   const storageResp = await fetch(signedUrl);
+
   if (!storageResp.ok) {
-    return null;
+    throwResponseError(storageResp);
```

Comment:
This didn't throw before, right? Is it caught later?

Reply:
Yea, every implementation of `fetchWork` that we have right now wraps the function call in a `try...catch` block.